### PR TITLE
Add test for ingress-custom-grpc-health-check

### PR DESCRIPTION
**Pull request by sawsa307**
* Update SSL policy name in the recipe README and yaml to avoid naming collision.
* Add a helper function to check if a string has the expected pattern count.
* Update instance image version since the official docker install script doesn't work for debian-11.
* Add test for ingress-grpc recipe. The traffic is validated by running a grpc-client image on a docker and send grpc request to the load balancer IP.

### CODE REVIEW

File: ingress/single-cluster/ingress-custom-grpc-health-check/run-test.sh [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+                docker.io/salrashid123/grpc_app /grpc_client \
+                --host=grpc.domain.com:443 \
+                --tlsCert /certs/CA_crt.pem  \
+                --servername grpc.domain.com --repeat ${repeating} -skipHealthCheck"
+
+    if [[ ! -z "${instance}" ]]; then
+        eval_cmd="gcloud compute ssh "${instance}" --zone="${zone}" -- \
+                    '{ docker --version || \
+                        (curl -fsSL https://get.docker.com -o get-docker.sh && sudo sh get-docker.sh); } > /dev/null && \
+                     sudo ${eval_cmd}'"
```
Comment (bowei) [inline ingress/single-cluster/ingress-custom-grpc-health-check/run-test.sh:67]: Is there a way to do this without having to install docker? Is there a debian package that offers a grpc_client?

File: ingress/single-cluster/ingress-custom-grpc-health-check/run-test.sh [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+check_ingress_grpc_response() {
+    local vip repeating pattern instance eval_cmd ATTEMPT
+    vip="$1"
+    repeating="$2"
+    pattern="$3"
+    instance="${4:-}"
+    eval_cmd="docker run \
+                --add-host grpc.domain.com:${vip} \
+                -t \
+                docker.io/salrashid123/grpc_app /grpc_client \
```
Comment (bowei) [inline ingress/single-cluster/ingress-custom-grpc-health-check/run-test.sh:58]: we should not use random containers on the Internet in our tests.

Comment (bowei) [reply ingress/single-cluster/ingress-custom-grpc-health-check/run-test.sh:58]: It is a security and reliability issue

Comment (sawsa307) [reply ingress/single-cluster/ingress-custom-grpc-health-check/run-test.sh:58]: This is described in the README so I follow within the test. 
I'll look for ways to validate using an official image for grpc_client or using grpcurl.

**Comment by sawsa307** [comment]
Closing this since it is not maintainable to use custom hosted image. Will follow up with test author to update it.

**Pull request closed without merging**